### PR TITLE
[Namespace]: Simplify sync_d functions to use higher order

### PR DESCRIPTION
**Pull request by SuvarnaMeenakshi**
functions.

Signed-off-by: SuvarnaMeenakshi <sumeenak@microsoft.com>

**- What I did**
Namespace class was added to support getting data from multiple namespaces.
This class includes some functions which get data from multiple namespaces and combine them.
This PR is to simplify the function of getting from separate functions and merge them.

**- How I did it**
Write a higher order function to invoke individual function to retrieve data from a single namespace.
The higher order function will retrieve the data and combine.

**- How to verify it**
No change in single and multi-asic snmp walk outputs.
Only implementation changes.
Make sure snmp walk outputs before change and after change are the same.

**- Description for the changelog**

### CODE REVIEW

File: src/sonic_ax_impl/mibs/__init__.py [hunk 1]
```diff
@@ -554,82 +554,23 @@ def get_non_host_dbs(dbs):
...
-        return port_queues_map, queue_stat_map, port_queue_list_map
+            ns_tuple = per_namespace_func(db_conn)
+            for idx in range(len(ns_tuple)):
+                if idx not in result_map:
+                    result_map[idx] = ns_tuple[idx]
+                else:
+                    result_map[idx].update(ns_tuple[idx])
+        for idx, ns_tuple_dict in result_map.items():
+            result_list.append(ns_tuple_dict)
+        return(result_list)
```
Comment (qiluo-msft) [inline src/sonic_ax_impl/mibs/__init__.py:573]:
```suggestion
        return result_list
```

File: src/sonic_ax_impl/mibs/__init__.py [hunk 2]
```diff
@@ -554,82 +554,23 @@ def get_non_host_dbs(dbs):
...
-
-        """
-        all_ns_db - will have db_conn to all namespace DBs and
-        global db. First db in the list is global db.
-        Ignore first global db to get queue tables if there
-        are multiple namespaces.
-        """
-        Namespace.connect_namespace_dbs(dbs)
+    def get_sync_d_from_all_namespace(per_namespace_func, dbs):
+        # return list of dictionaries retrieved from per namespace functions
```
Comment (qiluo-msft) [inline src/sonic_ax_impl/mibs/__init__.py:559]: >return list of dictionaries [](start = 10, length = 27)

You did not return list of dictionaries

Comment (SuvarnaMeenakshi) [reply src/sonic_ax_impl/mibs/__init__.py:559]: we return list of dictionaries, the dictionaries will be the same as return from individual sync_d functions, just that they are merged together.

Comment (SuvarnaMeenakshi) [reply src/sonic_ax_impl/mibs/__init__.py:559]: Probably, we can say "return tuple of dictionaries"

File: src/sonic_ax_impl/mibs/__init__.py (hunk 1 above)
Comment (SuvarnaMeenakshi) [reply src/sonic_ax_impl/mibs/__init__.py]: Updated as per comment.

File: src/sonic_ax_impl/mibs/__init__.py (hunk 2 above)
Comment (SuvarnaMeenakshi) [reply src/sonic_ax_impl/mibs/__init__.py]: Updated as per review comment.

**Pull request merged**